compiler/rustc_lexer/src/cursor.rs

```diff
@@ -81,4 +81,13 @@ impl<'a> Cursor<'a> {
 
         Some(c)
     }
+
+    /// Eats symbols while predicate returns true or until the end of file is reached.
+    pub(crate) fn eat_while(&mut self, mut predicate: impl FnMut(char) -> bool) {
+        // It was tried making optimized version of this for eg. line comments, but
```

Comment:
```suggestion
        // There was an attempt to make an optimized version of this for e.g. line comments, but
```

Reply:
That would have worked to.
But sorry, I saw this to late.